modules/Movecraft/src/main/java/net/countercraft/movecraft/async/AsyncManager.java

```diff
@@ -642,6 +663,12 @@ public CraftStatus checkCraftStatus(@NotNull Craft craft) {
         if (totalNonNegligibleBlocks == 0)
             isSinking = true;
 
+        if (craft instanceof BaseCraft) {
+            BaseCraft b = ((BaseCraft) craft);
+            b.updateMaterials(materials);
+            b.setTotalFuel(fuel);
+        }
+
```

Comment:
Like I said in the PR from ccorp, we should aim to use `BaseCraft` minimally and provide as much as we can in `Craft`.

Reply:
After reading further, you should be able to remove the `instanceof` check here.